src/robot/model/modelobject.py

```diff
@@ -26,4 +26,33 @@ def __unicode__(self):
     def __repr__(self):
         return repr(str(self))
 
+    def __setstate__(self, state):
+        """Customize the attribute updating when using `copy.copy` or `copy.deepcopy`
+        """
+
+        # We should consider the state format per pickle protocol version
+        # Refer to: https://www.python.org/dev/peps/pep-0307
+        if isinstance(state, tuple) and len(state) == 2:
+            dictstate, slotstate = state
+        else:
+            dictstate = state
+            slotstate = None
+
+        if dictstate is not None:
+            self.__dict__.update(dictstate)
+
+        def get_setter_name(x):
+            return '_setter__' + x
+
+        if slotstate is not None:
+            for k, v in slotstate.iteritems():
+                setter_name = get_setter_name(k)
+                # If we have defined the attribute both in __slots__ and with @setter,
+                # we just need to set the setter attribute.
+                if setter_name in slotstate:
+                    continue
+
+                setattr(self, k, v)
+
+
```

Comment:
Our code conventions say empty rows shouldn't generally be used inside functions/methods and there shouldn't be empty rows at the end.